llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/ObfuscateI8AddNewPassManager.cpp

```diff
@@ -0,0 +1,77 @@
+#include "llvm/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager.h"
+
+#include "llvm/Pass.h"
+
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+
+
+#include "llvm/Support/raw_ostream.h"
+
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+
+using namespace llvm;
+
+bool isBinaryAdd(Instruction const* instruction) {
+    return instruction->getOpcode() == Instruction::Add && instruction->getNumOperands() == 2;
+}
+
+bool isI8(Value const* operand) {
+    return (operand->getType()->isIntegerTy()) && (operand->getType()->getIntegerBitWidth() == 8);
+}
+
+bool withI8(Instruction const* instruction) {
+	auto const& operands = instruction->operands();
+    return std::any_of(operands.begin(), operands.end(),
+                       [](auto const& operand) { return isI8(operand); });
```

Comment:
It is easier to check bitwidth of operation itself:

```
binOp->getType()->isIntegerTy() && binOp->getType()->getIntegerBitWidth() == 8
```

Reply:
You still check operands, not operation

Reply:
Did it in 1a3cfa3a4d32dd7a43a75a0c965c6cee9bfb0727